emma-language/src/main/scala/org/emmalanguage/compiler/lang/core/DCE.scala

```diff
@@ -40,11 +42,19 @@ private[core] trait DCE extends Common {
       api.BottomUp.withValUses.transformWithSyn {
         case Attr(let @ core.Let(vals, defs, expr), _, _, syn) =>
           def refs(tree: u.Tree) = syn(tree).head.keySet
-          val liveRefs = vals.foldRight(refs(expr)) {
+
+          // Gather refs in DefDefs.
+          val refsInDefs = ArrayBuffer[u.TermSymbol]()
```

Comment:
No need to use a mutable buffer here. You can just do:
`val refsInDefs: Set[u.TermSymbol] = defs.flatMap(refs)(breakOut)`